stakingModel/test/GoodMarketMaker.test.js

```diff
@@ -60,16 +51,109 @@ contract(
       expect(expectedTotalMinted).to.be.equal(10000); //1k GD since price is 0.0001 cdai for 1 gd
       expect(toMint.toString()).to.be.equal(
         (expectedTotalMinted * 100).toString()
       ); //add 2 decimals precision
     });
 
+    it("should be able to calculate and update bonding curve gd balance based on oncoming cDAI and the price stays the same", async () => {
+      const priceBefore = await marketMaker.currentPrice(cDAI.address);
+      await marketMaker.mintInterest(
+        cDAI.address,
+        web3.utils.numberToHex(1e8)
+      );
+      let reserveToken = await marketMaker.reserveTokens(cDAI.address);
+      let gdSupply = reserveToken.gdSupply;
+      let reserveSupply = reserveToken.reserveSupply;
+      let reserveRatio = reserveToken.reserveRatio;
+      await marketMaker.initializeToken(
+        cDAI.address,
+        gdSupply.toString(),
+        (reserveSupply.toString()),
+        reserveRatio.toString()
+      );
+      expect((Math.floor((await marketMaker.currentPrice(cDAI.address)) / 100)).toString()).to.be.equal((Math.floor(priceBefore.toNumber() / 100)).toString());
+    });
+
+    it("should be able to update the reserve ratio only by the owner", async () => {
+      let error = await marketMaker.expandReserveRatio(
+                    cDAI.address, {
+                      from: staker
+                    }
+                  ).catch(e => e);
+      expect(error.message).not.to.be.empty;
+    });
+
+    it("should be able to update the bonding curve only by the owner", async () => {
+      let error = await marketMaker.mintInterest(
+                    cDAI.address,
+                    web3.utils.numberToHex(1e8), {
+                      from: staker
+                    }
+                  ).catch(e => e);
+      expect(error.message).not.to.be.empty;
+    });
+
+    it("should be able to update the bonding curve only by the owner", async () => {
+      let error = await marketMaker.mintExpansion(
+                    cDAI.address,
+                    web3.utils.numberToHex(1e8), {
+                      from: staker
+                    }
+                  ).catch(e => e);
+      expect(error.message).not.to.be.empty;
+    });
+
+    it("should be able to calculate minted gd based on expansion of reserve ratio, the price stays the same", async () => {
+      let reserveTokenBefore = await marketMaker.reserveTokens(cDAI.address);
+      let gdSupplyBefore = reserveTokenBefore.gdSupply;
+      let reserveSupplyBefore = reserveTokenBefore.reserveSupply;
+      let reserveRatioBefore = reserveTokenBefore.reserveRatio;
+      const priceBefore = await marketMaker.currentPrice(cDAI.address);
+      const toMint = await marketMaker.calculateMintExpansion(cDAI.address);
+      const newRR = await marketMaker.calculateNewReserveRatio(cDAI.address);
+      expect(reserveRatioBefore.toString()).not.to.be.equal(newRR.toString());
+      await marketMaker.initializeToken(
```

Comment:
you need to call mint and not initializeToken

Reply:
ohh this is just a calculation... but why initialize then?

Reply:
@sirpy 
to initialize the reserve with the new reserve ratio and the new gd supply to check the correctness of the calculation. the next test is similar, it checks the `mintExpansion` which also call to `calculateMintExpansion`.